components/script/textinput.rs

```diff
@@ -468,6 +471,86 @@ impl<T: ClipboardProvider> TextInput<T> {
         self.selection_begin = None;
     }
 
+    pub fn adjust_horizontal_by_word(&mut self, direction: Direction, select: Selection) {
+        if self.adjust_selection_for_horizontal_change(direction, select) {
+            return
+        }
+        let shift_increment: isize =  {
+            let input: &str;
+            match direction {
+                Direction::Backward => {
+                    let remaining = self.edit_point.index;
+                    let current_line = self.edit_point.line;
+                    let mut newline_adjustment = 0;
+                    if remaining == 0 && current_line > 0 {
+                        input = &self
+                            .lines[current_line-1];
+                        newline_adjustment = 1;
+                    } else {
+                        input = &self
+                            .lines[current_line]
+                            [..remaining];
+                    }
+
+                    match input
+                        .unicode_words()
+                        .last() {
+                            None => 0,
+                            Some(i) => {
+                                - (input.len() as isize + newline_adjustment as isize
+                                - input.rfind(i).expect("no words to the left on this line") as isize)
```

Comment:
Is there a way to avoid using `rfind` here? We're finding the last word, then searching back for it, which is less efficient than it could be.